TempleEight/docs/guide/business-logic.md

```diff
@@ -76,21 +79,228 @@ import (
 )
 
 func (env *env) setup(router *mux.Router) {
-	env.hook.BeforeCreate(beforeCreateHook)
+	env.hook.BeforeCreate(ourCustomBeforeHook)
 }
 
-func beforeCreateHook(env *env, req createExampleServiceRequest, input *dao.CreateExampleServiceInput) *HookError {
+func ourCustomBeforeHook(env *env, req createExampleServiceRequest, input *dao.CreateExampleServiceInput) *HookError {
 	return nil
 }
 ```
 
-// TODO
 ## Modifying the DAO request
+Now that we have defined our custom hook, we can start populating it with additional logic.
+To start, we will show how you can modify the DAO request.
+The DAO provides a common interface to access a backing store, without directly exposing the implementation details of doing so.
+More information about this can be found in [TODO](the-go-section.md)
+
+The custom hook we defined takes 3 parameters:
+- the environment, `env`
+- the request provided by the user, `req`
+- the input to the datastore request, `input`
+
+The input object is passed as a pointer, which gets directly passed to the DAO after the hook invocation and will be used as the query for the datastore.
+This means if we update any attributes of this object, the data store call will be updated too.
+In the following example, we modify all `Create` requests to the datastore so that each `foo` property contains the string "Hello, World!".
+
+```go
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/temple/tutorial/example-service/dao"
+)
+
+func (env *env) setup(router *mux.Router) {
+	env.hook.BeforeCreate(ourCustomBeforeHook)
+}
+
+func ourCustomBeforeHook(env *env, req createExampleServiceRequest, input *dao.CreateExampleServiceInput) *HookError {
+	input.Foo = "Hello, world!"
+	return nil
+}
+```
+If we perform some example requests, we see that our hook updates the object that is stored, irrespective of what was passed in the request:
+
+```bash
+# Create a new object
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "abcd", "bar": 10}'
+{"id":"43cc65f5-823c-11ea-9dc4-0242ac180003","foo":"Hello, world!","bar":10}
+
+# Retrieve that same object 
+❯❯❯ curl -X GET $KONG_ENTRY/api/example-service/43cc65f5-823c-11ea-9dc4-0242ac180003
+{"id":"43cc65f5-823c-11ea-9dc4-0242ac180003","foo":"Hello, world!","bar":10}
+```
+
+
+## Conditionally updating the datastore input
+As well as updating the datastore request, we could also use the user's request to conditionally update certain fields:
+```go
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/temple/tutorial/example-service/dao"
+)
+
+func (env *env) setup(router *mux.Router) {
+	env.hook.BeforeCreate(ourCustomBeforeHook)
+}
+
+func ourCustomBeforeHook(env *env, req createExampleServiceRequest, input *dao.CreateExampleServiceInput) *HookError {
+	if (req.Bar == 5) {
+		input.Foo = "Hello, world!"
+	}
+	return nil
+}
+```
+
+This should only update the value of `Foo` if the value 5 is passed for bar: 
```

Comment:
```suggestion
This will only update the value of `Foo` if the value 5 is passed for bar: 
```

Reply:
Thanks, have removed all other `should` usages too